src/tree/containerizedFunctionApp/FunctionsTreeItem.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type FunctionEnvelope, type Site } from "@azure/arm-appservice";
+import { createWebSiteClient } from "@microsoft/vscode-azext-azureappservice";
+import { uiUtils } from "@microsoft/vscode-azext-azureutils";
+import { nonNullProp, nonNullValueAndProp, type AzExtTreeItem, type IActionContext } from "@microsoft/vscode-azext-utils";
+import * as retry from 'p-retry';
+import { ParsedFunctionJson } from "../../funcConfig/function";
+import { localize } from "../../localize";
+import { FunctionsTreeItemBase } from "../FunctionsTreeItemBase";
+import { getFunctionNameFromId } from "../remoteProject/RemoteFunctionTreeItem";
+import { type ContainerTreeItem } from "./ContainerTreeItem";
+import { FunctionItem } from "./FunctionItem";
+import { FunctionTreeItem } from "./FunctionTreeItem";
+
+export class FunctionsTreeItem extends FunctionsTreeItemBase {
+    public isReadOnly: boolean = true;
+    private _nextLink: string | undefined;
+
+    constructor(public readonly parent: ContainerTreeItem, public readonly site: Site) {
+        super(parent);
+    }
+
+    public static async createFunctionsTreeItem(context: IActionContext, parent: ContainerTreeItem): Promise<FunctionsTreeItem> {
+        const ti: FunctionsTreeItem = new FunctionsTreeItem(parent, parent.site);
+        // initialize
+        await ti.refreshImpl(context);
+        return ti;
+    }
+
+    public hasMoreChildrenImpl(): boolean {
+        return !!this._nextLink;
+    }
+
+    public async refreshImpl(context: IActionContext): Promise<void> {
+        this.isReadOnly = await this.parent.isReadOnly(context);
+    }
+
+    public async loadMoreChildrenImpl(clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
+        if (clearCache) {
+            this._nextLink = undefined;
+        }
+        /*
+            Related to issue: https://github.com/microsoft/vscode-azurefunctions/issues/3179
```

Comment:
Do you know if you actually need all of this extra retry logic here? I'm not sure if the containers have the same "BadGateway" errors and/or have a warm-up function.

It doesn't really hurt to have it, but it'd be nice to be able to clean up this file a bit.

Reply:
Hmm not totally sure if we need it and also not sure about the errors/warm up function but I can take a look into it.